x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_page/hooks/use_rule_type_ids_by_feature_id.ts

```diff
@@ -33,10 +33,14 @@ export const useRuleTypeIdsByFeatureId = (ruleTypesIndex: RuleTypeIndex) =>
     const map = Array.from(ruleTypesIndex.entries()).reduce<RuleTypeIdsByFeatureId<Set<string>>>(
       (types, [key, value]) => {
         let producer = value.producer as keyof RuleTypeIdsByFeatureId;
-        // All o11y apps are listed under 'observability' to create a grouped filter
-        producer = observabilityFeatureIds.includes(producer)
-          ? AlertConsumers.OBSERVABILITY
-          : producer;
+        // Some o11y apps are listed under 'observability' to create a grouped filter
+        if (observabilityFeatureIds.includes(producer)) {
+          producer = AlertConsumers.OBSERVABILITY;
+        }
+        // Stack includes ML in this context
+        if (stackFeatureIds.includes(producer)) {
+          producer = AlertConsumers.STACK_ALERTS;
+        }
```

Comment:
Do we only want to do this in the context of the alerts table? Have we discussed perhaps changing ML rule types to use `stackAlerts` as it's producer for all applications?

Reply:
cc @shanisagiv1